File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        res = 0
        j = 1
        k = 0
        for i in st:
            if i - 1 not in st:
                j = 1
                k = i + 1
                while k in st:
                    k += 1
                    j += 1
                res = max(res, j)
        return res
                
            
                

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::unordered_map<int, int> mp;
        int res = 0;
        for (int i: nums)
            mp[i] = 1;
        int j;
        std::unordered_map<int, int>::iterator tem;
        for (std::unordered_map<int, int>::iterator pr = mp.begin(); pr != mp.end(); pr++)
        {
            tem = mp.find(pr->first + 1);
            if (tem != mp.end())
            {
                j = tem->first;
                while (tem != mp.end())
                {
                    res += tem->second;
                    mp.erase(j);
                    j++;
                    tem = mp.find(j);
                }
            }
            pr->second += res;
            res = 0;
        }
        for (auto t: mp)
            if (t.second > res)
                res = t.second;
        return res;    
    }
};